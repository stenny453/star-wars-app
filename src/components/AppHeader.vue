<template>
  <div class="main-header">
    <vs-navbar
      v-model="activeItem"
      class="nabarx"
      active-text-color="#5b3cc4"
      type="flat"
    >
      <div class="navbar-item">
        <div slot="title">
          <vs-navbar-title>
            <span class="header-title">
              {{ title }}
            </span>
          </vs-navbar-title>
        </div>

        <vs-navbar-item
          v-for="path in paths"
          :key="path.index"
          :index="path.index"
        >
          <RouterLink :to="path.link">
            {{ path.name }}
          </RouterLink>
        </vs-navbar-item>
      </div>

      <div class="navbar-search">
        <vs-input
          v-model="search"
          icon="search"
          placeholder="Search"
        />
      </div>
    </vs-navbar>
  </div>
</template>

<script setup>
import { ref, toRefs } from 'vue';
import { RouterLink } from 'vue-router';

const activeItem = ref(0);

const search = ref('');

// eslint-disable-next-line no-undef
const props = defineProps({
  title: {
    type: String,
    required: true,
  },
  paths: {
    type: Array,
    required: true,
  },
});

const { title, paths } = toRefs(props);

</script>

<style>
  .main-header {
    padding: 10px;
    position: sticky;
    z-index: 2;
    top: 0;
    background-color: white;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1), 0 4px 8px rgba(0, 0, 0, 0.1);
  }

  .title {
    color: #5b3cc4;
    font-weight: bold;
  }

  .vs-con-items {
    width: 100%;
    justify-content: space-between !important;
    min-height: 60px;
    padding-right: 10px;
  }

  .navbar-item {
    display: flex;
    align-items: center;
  }

  .vs-input--icon {
    top: 4px !important;
  }

  @media screen and (max-width: 800px) {
    .navbar-item {
      flex-direction: column;
    }

    .navbar-search {
      width: 100%;
      display: flex;
      justify-content: center;
    }
  }
</style>
