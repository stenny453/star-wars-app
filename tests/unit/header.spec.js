/* eslint-disable no-undef */
/* eslint-disable import/no-unresolved */

import { shallowMount } from '@vue/test-utils';
import AppHeader from '@/components/AppHeader.vue';

describe('AppHeader.vue', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(AppHeader, {
      props: {
        title: 'My App',
        paths: [
          {
            index: 0,
            link: '/',
            name: 'Movies',
          },
          {
            index: 1,
            link: '/classic-movies',
            name: 'Classics',
          },
        ],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('props well received', () => {
    const wrapper = shallowMount(AppHeader, {
      props: {
        title: 'My App',
        paths: [
          {
            index: 0,
            link: '/',
            name: 'Movies',
          },
          {
            index: 1,
            link: '/classic-movies',
            name: 'Classics',
          },
        ],
      },
    });

    // check if props well passed
    expect(wrapper.props().title).toBe('My App');

    // check if paths well passed
    expect(wrapper.props().paths).toEqual([
      {
        index: 0,
        link: '/',
        name: 'Movies',
      },
      {
        index: 1,
        link: '/classic-movies',
        name: 'Classics',
      },
    ]);

    // check if title of app is equal to props title
    expect(wrapper.find('.header-title').text()).toBe('My App');
  });
});
