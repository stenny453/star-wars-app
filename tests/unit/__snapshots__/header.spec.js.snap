// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader.vue should render correctly 1`] = `
<div class="main-header">
  <vs-navbar modelvalue="0" class="nabarx" active-text-color="#5b3cc4" type="flat">
    <div class="navbar-item">
      <div slot="title">
        <vs-navbar-title><span class="header-title">My App</span></vs-navbar-title>
      </div>
      <vs-navbar-item index="0">
        <router-link-stub to="/" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
      </vs-navbar-item>
      <vs-navbar-item index="1">
        <router-link-stub to="/classic-movies" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
      </vs-navbar-item>
    </div>
    <div class="navbar-search">
      <vs-input modelvalue="" icon="search" placeholder="Search"></vs-input>
    </div>
  </vs-navbar>
</div>
`;
